pkg/controller/kibana/config/settings.go

```diff
@@ -73,12 +90,38 @@ func NewConfigSettings(client k8s.Client, kb kbv1.Kibana, versionSpecificCfg *se
 	return CanonicalConfig{cfg}, nil
 }
 
+// getExistingConfig retrieves the canonical config for a given Kibana, if one exists
+func getExistingConfig(client k8s.Client, kb kbv1.Kibana) *settings.CanonicalConfig {
+	var secret corev1.Secret
+	err := client.Get(types.NamespacedName{Name: SecretName(kb), Namespace: kb.Namespace}, &secret)
+	if err != nil && apierrors.IsNotFound(err) {
+		log.V(1).Info("Kibana config secret does not exist", "kibana_namespace", kb.Namespace, "kibana_name", kb.Name)
+		return nil
+	} else if err != nil && !apierrors.IsNotFound(err) {
+		log.Error(err, "Error retrieving kibana config secret", "kibana_namespace", kb.Namespace, "kibana_name", kb.Name)
+		return nil
+	}
+	rawCfg, exists := secret.Data[SettingsFilename]
+	if !exists {
+		log.Error(nil, "No kibana config file in secret", "secret_namespace", secret.Namespace, "secret_name", secret.Name, "key", SettingsFilename)
+		return nil
+	}
+	cfg, err := settings.ParseConfig(rawCfg)
+	if err != nil {
+		log.Error(err, "Error parsing existing kibana config in secret", "secret_namespace", secret.Namespace, "secret_name", secret.Name, "key", SettingsFilename)
+		return nil
+	}
+	return cfg
+}
+
 func baseSettings(kb kbv1.Kibana) map[string]interface{} {
 	return map[string]interface{}{
 		ServerName:         kb.Name,
 		ServerHost:         "0",
 		ElasticSearchHosts: []string{kb.AssociationConf().GetURL()},
 		XpackMonitoringUiContainerElasticsearchEnabled: true,
+		// this will get overriden if one already exists or is specified by the user
+		XpackSecurityEncryptionKey: rand.String(64),
```

Comment:
Based on what you said in the PR top post , I would also add something like:
`// if the user specifies one in secure settings, we still generate one in the Kibana configuration file, but the secure settings one takes precedence`

Reply:
It's actually not clear to me which one takes precedence -- from what I could see there's no way to introspect what encryption key it decides to use, but I could definitely be missing it. I was able to verify that it didn't error out at least and did use _an_ encryption key.

What I meant is that it doesn't get overridden even if they just put it in spec. I'll add a test for that case though

Reply:
I think we need to make sure if the user sets an encryption key in their secure settings, that's the one that will be used and not our randomly generated one. We may need to inspect the secure settings content?

Reply:
Let me see if I can find out which takes precedence. If the secure setting does not take precedence we may want to do that for all values defined there.